aexpect/ops.py

```diff
@@ -0,0 +1,439 @@
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str filename: full path of file
+    :returns: file contents
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if cat command fails
+
+    Should only be used for very short files without tabs or other fancy
+    contents. Otherwise better download file or use some other method.
+    """
+    cmd = 'cat ' + quote(filename)
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError(f'Failed to cat {filename}: {output}')
+    return output
+
+
+def make_tempdir(session, template="tmp.XXXXXXXXXX"):
+    """
+    Create a temporary directory in a safe way through a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str template: directory name template as used in the standard command
+    :returns: name of temporary directory
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if mktemp returns non-zero exit status
+
+    Calls `mktemp -d`, refer to `man` for more info.
+    """
+    cmd = f'mktemp -d {template}'
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError(f'Failed to create temporary directory: {output}')
+    return output
+
+
+def make_tempfile(session, template="tmp.XXXXXXXXXX"):
+    """
+    Create a temporary file in a safe way through a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str template: file name template as used in the standard command
+    :returns: name of temporary file
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if mktemp returns non-zero exit status
+
+    Calls `mktemp`, refer to `man` for more info.
+    """
+    cmd = f'mktemp {template}'
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError(f'Failed to create temporary file: {output}')
+    return output
+
+
+def move(session, source, target):
+    """
+    Move file or directory from source to target / rename source to target.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str source: full path of existing file or directory
+    :param str target: full path of new file or directory
+    :raises: :py:class:`RuntimeError` if mv returns non-zero exit status
+
+    Calls `mv source target` through a session. See `man mv` for what source
+    and target can be and what behavior to expect.
+    """
+    cmd = f'mv {quote(source)} {quote(target)}'
+    status, output = session.cmd_status_output(cmd)
+    if status != 0:
+        raise RuntimeError(f"Failed to move {source} to {target}: "
+                           f"{output.strip() if output else ''}")
+
+
+def copy(session, source, target):
```

Comment:
Here it'd really make sense to allow `flags` (eg. for `-r`)

Reply:
Alright, let's cover at least the few most obvious choices here